Implement Hsql XADatasource and Datasource as ecm components.

kalina91 opened the pull request.

Commit:
Implement Hsql XADatasource and Datasource as ecm components.

balazs-zsoldos commented on .gitignore:
```diff
@@ -0,0 +1,8 @@
+/.checkstyle
+/.pmdruleset.xml
+/.pmd
+/.fbExcludeFilterFile
```
missing new line from end of file

balazs-zsoldos commented on pom.xml:
```diff
@@ -0,0 +1,134 @@
+
+  <packaging>bundle</packaging>
+  <name>Everit - HyperSQL ECM</name>
+  <description>Configurable ECM based component that makes it possible to instantiate and register JtdsDataSource and register it as an OSGi service based on the interfaces it implements.</description>
```
Jtds datasource?

balazs-zsoldos commented on pom.xml:
```diff
@@ -0,0 +1,134 @@
+              *
+            </Import-Package>
+            <Export-Package>
+              org.everit.persistence.jtds.ecm
```
jtds?

balazs-zsoldos commented on pom.xml:
```diff
@@ -0,0 +1,134 @@
+    </dependency>
+  </dependencies>
+
+</project>
```
no new line at end of file

balazs-zsoldos commented on src/main/java/org/everit/persistence/hsqldb/ecm/internal/AbstractHsqlDatasourceComponent.java:
```diff
@@ -0,0 +1,71 @@
+
+  protected String user;
+
+  @StringAttribute(attributeId = HsqlDataSourceComponentConstants.ATTR_PASSWORD,
```
Password should be PasswordAttribute, not StringAttribute

balazs-zsoldos commented on src/main/java/org/everit/persistence/hsqldb/ecm/internal/HsqlDataSourceComponent.java:
```diff
@@ -0,0 +1,78 @@
+    }
+  }
+
+  private void updatePropertiesAToD(final JDBCDataSource jdbcDataSource) {
```
This could be in parent class with the CommonJDBCDataSource as parameter as it is the parent of XA and non-XA datasources

balazs-zsoldos commented on src/main/java/org/everit/persistence/hsqldb/ecm/internal/HsqlDataSourceComponent.java:
```diff
@@ -0,0 +1,78 @@
+    Dictionary<String, Object> serviceProperties =
+        new Hashtable<>(componentContext.getProperties());
+    serviceRegistration = componentContext.registerService(
+        new String[] { DataSource.class.getName() },
```
Add the implementation class to the list of interfaces and if useful, other interfaces that the class implements, too.

balazs-zsoldos commented on src/main/java/org/everit/persistence/hsqldb/ecm/internal/HsqlXADataSourceComponent.java:
```diff
@@ -0,0 +1,83 @@
+    Dictionary<String, Object> serviceProperties =
+        new Hashtable<>(componentContext.getProperties());
+    serviceRegistration = componentContext.registerService(
+        new String[] { XADataSource.class.getName() },
```
Add the implementation class to the list of interfaces and if useful, other interfaces that the class implements, too.

Commit:
Do changes advised by Balázs Zsoldos.

Publish HsqlXADataSourceComponent and HsqlDataSourceComponent as
CommonDataSource.
Extract updatePropertiesAToD method to AbstractHsqlDatasourceComponent.

kalina91 commented:
I updated the code, please review again.
